onnxruntime/core/providers/rocm/rocm_profiler.cc

```diff
@@ -7,231 +7,83 @@
-                                            {"grid_z", std::to_string(item.gridZ)},
-                                            {"block_x", std::to_string(item.workgroupX)},
-                                            {"block_y", std::to_string(item.workgroupY)},
-                                            {"block_z", std::to_string(item.workgroupZ)},
-                                            };
-    addEventRecord(item, profiling_start, args, event_map);
-    kernelLaunches[item.id] = &item;
-  }
+    // find the last event with the same timestamp.
+    while (event_iter != event_end && event_iter->ts == ts && (event_iter + 1)->ts == ts) {
+      ++event_iter;
+    }
 
-  // Async Ops - e.g. "Kernel"
-
-  for (auto& buffer : *d->gpuTraceBuffers_) {
-    const roctracer_record_t* record = (const roctracer_record_t*)(buffer.data_);
-    const roctracer_record_t* end_record = (const roctracer_record_t*)(buffer.data_ + buffer.validSize_);
-    while (record < end_record) {
-      std::unordered_map<std::string, std::string> args;
-      std::string name = roctracer_op_string(record->domain, record->op, record->kind);
-
-      // Add kernel args if we have them
-      auto kit = kernelLaunches.find(record->correlation_id);
-      if (kit != kernelLaunches.end()) {
-        auto &item = *(*kit).second;
-        snprintf(buff, BSIZE, "%p", item.stream);
-        args["stream"] = std::string(buff);
-        args["grid_x"] = std::to_string(item.gridX);
-        args["grid_y"] = std::to_string(item.gridY);
-        args["grid_z"] = std::to_string(item.gridZ);
-        args["block_x"] = std::to_string(item.workgroupX);
-        args["block_y"] = std::to_string(item.workgroupY);
-        args["block_z"] = std::to_string(item.workgroupZ);
-        if (item.functionAddr != nullptr) {
-          name = demangle(hipKernelNameRefByPtr(item.functionAddr, item.stream)).c_str();
-        }
-        else if (item.function != nullptr) {
-          name = demangle(hipKernelNameRef(item.function)).c_str();
-        }
-      }
+    if (event_iter != event_end && event_iter->ts == ts) {
+      uint64_t increment = 1;
+      for (auto& evt : map_iter.second) {
+        evt.args["op_name"] = event_iter->args["op_name"];
 
-      // Add copy args if we have them
-      auto cit = copyLaunches.find(record->correlation_id);
-      if (cit != copyLaunches.end()) {
-        auto &item = *(*cit).second;
-        snprintf(buff, BSIZE, "%p", item.stream);
-        args["stream"] = std::string(buff);
-        snprintf(buff, BSIZE, "%p", item.src);
-        args["dst"] = std::string(buff);
-        snprintf(buff, BSIZE, "%p", item.dst);
-        args["src"] = std::string(buff);
-        args["size"] = std::to_string(item.size);
-        args["kind"] = std::to_string(item.kind);
+        // roctracer timestamps don't use Jan 1 1970 as an epoch,
+        // not sure what epoch it uses, but we adjust the timestamp
```

Comment:
nit - better the grammar, either be with plural form to the end or never.

Reply:
Thanks, done!